File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        s = s.trim();
        StringBuilder reverse = new StringBuilder();
        int end = s.length();

        for (int i = s.length() - 1; i >= 0; i--) {
            if (s.charAt(i) == ' ') {
                reverse.append(s.substring(i + 1, end)).append(" ");
                while (i >= 0 && s.charAt(i) == ' ') i--; // Skip extra spaces
                end = i + 1;
            }
        }
        reverse.append(s.substring(0, end)); // Append the first word

        return reverse.toString();
    }
}

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        s = s.strip()
        
        reverse=""
        substring=""
        for i in range(len(s)-1, -1, -1):
            if s[i]==" ":
                if substring:
                    tempsub=substring[::-1]
                    reverse = reverse + tempsub+ " "
                    substring = ""
                
            else:
                substring+=s[i]
        if substring:     
            tempsub=substring[::-1]
            reverse = reverse + tempsub   
            
                
        return reverse
            
                
            
            